src/historywork/VerifyTxResultsWork.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+
+#include "util/TmpDir.h"
+#include "util/XDRStream.h"
+#include "work/BasicWork.h"
+#include "xdr/Stellar-types.h"
+
+namespace stellar
+{
+/*
+ * Verify transaction results for a checkpoint. This work requires
+ * downloaded ledger header and transaction result files.
+ * */
+class VerifyTxResultsWork : public BasicWork
+{
+    TmpDir const& mDownloadDir;
+    uint32_t const mCheckpoint;
+    TransactionHistoryResultEntry mTxResultEntry;
+    XDRInputFileStream mHdrIn;
+    XDRInputFileStream mResIn;
+    bool mDone{false};
+    asio::error_code mEc;
+
+    TransactionHistoryResultEntry getCurrentTxResultSet(uint32_t ledger);
+    bool verifyTxResultsOfCheckpoint();
+
+  public:
+    VerifyTxResultsWork(Application& app, TmpDir const& downloadDir,
+                        uint32_t checkpoint);
+
+  protected:
+    BasicWork::State onRun() override;
+    void onReset() override;
+
+    bool
+    onAbort() override
+    {
+        return true;
```

Comment:
I don't know if you can implement `abort` like this: this will cause callers to unwind even though verifications are happening in the background which may impact cleanup code, or retries in some weird way. I would imagine that you need to cause the worker to fail and wait for the failure to be posted back onto the main thread.